sdk/aqueduct/models/dag.py

```diff
@@ -457,6 +457,20 @@ def update_artifact_name(self, artifact_id: uuid.UUID, new_name: str) -> None:
         artifact.name = new_name
         artifact.explicitly_named = True
 
+    def update_artifact_should_persist(self, artifact_id: uuid.UUID, should_persist: bool) -> None:
+        self.must_get_artifact(artifact_id).should_persist = should_persist
+
+    def disable_snapshots_from_non_params_metrics_checks(self) -> None:
```

Comment:
I personally find this name a little confusing. I'd prefer if we kept the method more generic `disable_snapshots()` and added a nice docstring here indicating that parameters, metric and check artifacts are excluded.

Reply:
I actually prefer the naming to be self-descriptive esp. that here the underlying assumption is not clear to whoever scan through the caller codes.

![Screenshot 2023-06-05 at 2 21 06 PM](https://github.com/aqueducthq/aqueduct/assets/10411887/ddfeb6c9-cd0e-4dea-ba3d-20e1705c1fcb)

Reply:
how about `disable_snapshots_not_from_params_metrics_checks` ?

Reply:
I also prefer my methods to be meaningful, but the description needs to stop somewhere. In this case, it feels like we're leaking implementation details into the method name. Specifically, I'd rather not be listing things in my names (params, metrics, checks), since it's not scalable or very readable imo. Nbd if you want to keep it, it just jumped out at me.

Reply:
I'm more worried that `disable_snapshot` is misleading because we don't actually disable snapshots. I do agree that this name is too long and what you mentioned about listing stuff makes sense. Welcome to advices as long as we somehow indicate only some snapshots get disabled here :)

Reply:
Updated to `disable_sensitive_snapshots`. `sensitive` may confuse reader but that should lead them to the comment where we clarify things

Reply:
mm does `disable_op_snapshots` or `disable_op_output_snapshots` work? No strong opinions as long as we document it in the function.

Reply:
yeah sounds good, I guess I just don't want it look too harmless so that call reader would try to look at the docs, rather than skip over it with some wrong assumptions